### PR TITLE
remove assertEqualIgnoreTypes from test/distributions/test_distributions.py

### PR DESCRIPTION
**Pull request by liuk22**
See https://github.com/pytorch/pytorch/issues/38095

### CODE REVIEW

File: test/distributions/test_distributions.py [hunk 1]
```diff
@@ -890,13 +890,11 @@ def _check_enumerate_support(self, dist, examples):
             params = {k: torch.tensor(v) for k, v in params.items()}
             expected = torch.tensor(expected)
             d = dist(**params)
-            actual = d.enumerate_support(expand=False)
-            # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-            self.assertEqualIgnoreType(actual, expected)
+            actual = d.enumerate_support(expand=False)5
```
Comment (kit1980) [inline test/distributions/test_distributions.py:893]: Random "5" here for some reason.

**Review by kit1980 — COMMENTED** [review]
I've started CI tests run, let's see the results.

**Comment by liuk22** [comment]
@kit1980 all runs passing except for this one on MacOS M1, which appears to have many tests failing but they don't seem related, for example 

`[gw1] [ 15%] XFAIL test_ops.py::TestCommonCPU::test_out_fft_ihfftn_cpu_float32 ` 

Next steps? 

https://github.com/pytorch/pytorch/runs/7925703426?check_suite_focus=true

**Comment by kit1980** [comment]
> @kit1980 all runs passing except for this one on MacOS M1, which appears to have many tests failing but they don't seem related, for example
> 
> `[gw1] [ 15%] XFAIL test_ops.py::TestCommonCPU::test_out_fft_ihfftn_cpu_float32 `
> 
> Next steps?
> 
> https://github.com/pytorch/pytorch/runs/7925703426?check_suite_focus=true

XFAIL is not a problem - it means the test was expected to fail and failed.
The actual problem looks like "ERROR ENCOUNTERED WHEN UPLOADING TO SCRIBE".
I've re-started the workflow. If it fails again, we'll merge with force.

File: test/distributions/test_distributions.py [hunk 2]
```diff
@@ -1203,10 +1201,9 @@ def test_binomial_extreme_vals(self):
 
     def test_binomial_vectorized_count(self):
         set_rng_seed(1)  # see Note [Randomized statistical tests]
-        total_count = torch.tensor([[4, 7], [3, 8]])
+        total_count = torch.tensor([[4, 7], [3, 8]], dtype=torch.float64)
```
Comment (kit1980) [inline test/distributions/test_distributions.py:1204]: I don't understand why here and below it's float64. Locally for me Binomial and Multinomial sample() returns float32.

Comment (liuk22) [reply test/distributions/test_distributions.py:1204]: See this CI log line from the commit before I added that dtyping for float64: https://fburl.com/z6sdhva3

sample() returned float64 on several of the builds from that run.

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
LGTM!

**Pull request closed without merging**